app/Policies/AlbumPolicy.php

```diff
@@ -27,11 +27,16 @@ class AlbumPolicy
 
 	// constants to be used in GATE
 	public const IS_OWNER = 'isOwner';
+
+	public const CAN_SEE = 'canSee';
 	public const CAN_ACCESS = 'canAccess';
 	public const CAN_DOWNLOAD = 'canDownload';
+	public const CAN_UPLOAD = 'canUpload';
 	public const CAN_EDIT = 'canEdit';
-	public const IS_VISIBLE = 'isVisible';
 	public const CAN_EDIT_ID = 'canEditById';
+	public const CAN_SHARE_WITH_USERS = 'canShareWithUsers';
+	public const CAN_SHARE_BY_LINK = 'canShareByLink';
```

Comment:
Maybe we should rename that to something like `CAN_SEE_SHARE_LINKS` or `CAN_SEE_DIRECT_LINKS_FOR_SHARING`. The point is we cannot actually forbid someone to share links and the links are straightforward if you know the ID of the photo or album. This whole right is more or less only a convenience feature whether the user sees the button in the toolbar and has the option to easily copy the links.

I also thought that we already had agreed upon making this a global configuration options which controls the whole UI at once whether the button is shown in the toolbar or not. It seems to be an overkill to have this on a per-album/per-photo basis given the fact that this isn't a real security feature.